D/D.sublime-syntax

```diff
@@ -8,339 +8,2032 @@ file_extensions:
+        - match: '\b(unittest|assert)\b'
+          scope: keyword.d
+          set: version-condition-end
+        - match: '\b({{name}})\b'
+          scope: constant.other.d
+          set: version-condition-end
+        - match: '{{integer_lookahead}}'
+          set: [version-condition-end, integer]
+        - include: not-whitespace-illegal
+    - include: not-whitespace-illegal
+  version-condition-end:
+    - match: '\)'
+      scope: punctuation.section.parens.end.d
+      pop: true
+    - include: not-whitespace-illegal
+
+  debug-condition:
+    - match: '\('
+      scope: punctuation.section.parens.begin.d
+      set:
+        - match: '\b({{name}})\b'
+          scope: constant.other.d
+          set: version-condition-end
+        - match: '{{integer_lookahead}}'
+          set: [version-condition-end, integer]
+        - include: not-whitespace-illegal
+    - match: '(?=\S)'
+      pop: true
+
+  static-foreach-in:
+    - match: '\b(static)\s+(foreach|foreach_reverse)\b'
       captures:
-        1: punctuation.definition.comment.d
-  constant_placeholder:
-    - match: '(?i:%(\([a-z_]+\))?#?0?\-?[ ]?\+?([0-9]*|\*)(\.([0-9]*|\*))?[hL]?[a-z%])'
-      scope: constant.other.placeholder.d
-  function_assertions:
-    - match: '(in|out)\s*(?=\{)'
+        1: keyword.control.loop.d
+        2: keyword.control.loop.d
+      push: [optional-block-statement, foreach-arguments]
+
+  static-assert-in:
+    - match: '\b(static)\s+(assert)\b'
+      captures:
+        1: keyword.control.conditional.d
+        2: keyword.control.conditional.d
+      push: [expect-end-of-line, assert-arguments]
+
+  constructions-in:
+    - match: '{{construction_lookahead}}'
+      push: construction
+  construction:
+    # Post-blit constructor
+    - match: '\b(this)\s*(\()\s*(this)\s*(\))'
+      captures:
+        1: entity.name.function.post-blit.d
+        2: punctuation.section.group.begin.d
+        3: variable.language.d
+        4: punctuation.section.group.end.d
+      set: function-definition-after-arguments
+    - match: '\b(this)\s*(?=\(|$)'
+      captures:
+        1: entity.name.function.constructor.d
+        2: punctuation.section.group.begin.d
+      set: constructor-argument-list
+    - match: '(~\s*this)\s*(\()\s*(\))'
       captures:
-        1: keyword.control.statement.d
+        1: entity.name.function.destructor.d
+        2: punctuation.section.group.begin.d
+        3: punctuation.section.group.end.d
+      set: function-definition-after-arguments
+    - include: not-whitespace-illegal-pop
+  # This could either be function/template parameters or arguments. Assume
+  # arguments and switch to parameters when possible
+  constructor-argument-list:
+    - match: '\('
+      scope: punctuation.section.group.begin.d
+      set:
+        - match: '\)'
+          scope: punctuation.section.group.end.d
+          set: function-definition-after-first-argument-list
+        - match: '{{parameter_attribute_lookahead}}'
+          set
```

Comment:
One thing to look out for with constructs such as this is as the user is typing, when they've got `debug` typed in, ideally it should be highlighted the same as `debug =`. This may be the case here, but something to check for.

Reply:
At least in this case `version` and `debug` are other statements, so they'll be highlighted correctly. Though it might still be better to leave the `=` parsing to those statements instead.